experimental/beluga_demo_mh_amcl/include/beluga_demo_mh_amcl/particles_distribution.hpp

```diff
@@ -0,0 +1,182 @@
+// Copyright 2025 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MH_AMCL__PARTICLESDISTRIBUTION_HPP_
+#define MH_AMCL__PARTICLESDISTRIBUTION_HPP_
+
+#include <random>
+#include <vector>
+
+#include <beluga_ros/occupancy_grid.hpp>
+
+#include <geometry_msgs/msg/transform_stamped.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_lifecycle/lifecycle_node.hpp>
+#include <sensor_msgs/msg/laser_scan.hpp>
+#include <tf2/LinearMath/Transform.h>
+#include <tf2/transform_datatypes.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#include <tf2_ros/transform_listener.h>
+#include <visualization_msgs/msg/marker_array.hpp>
+
+#include "beluga_demo_mh_amcl/utils.hpp"
+
+namespace mh_amcl
+{
+
+  const double kInitialHypothesisQuality = 0.25;
+
+  /**
+   * @brief Structure representing a Particle, containing the pose, probability
+   * and hits (normalization of how well a particle aligns with the laser data).
+   * This 'hits' value is used as the quality of the particle.
+   */
+  struct Particle
+  {
+    Sophus::SE2d state;
+    beluga::Weight weight;
+    float hits;
+  };
+
+  /**
+   * ParticlesDistribution class
+   *
+   * Defines a set of particles (hypothesis). the core logic of the algorithm is
+   * performed in this class.
+   */
+  class ParticlesDistribution
```

Comment:
@JesusSilvaUtrera meta: I'm not super fond of the lifecycle'd distribution but OK